sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -2063,4 +2063,178 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
       )
     }
   }
+
+  def testFile(fileName: String): String = {
+    Thread.currentThread().getContextClassLoader.getResource(fileName).toString
+  }
+
+  test("json in UTF-16 with BOM") {
+    val fileName = "json-tests/utf16WithBOM.json"
+    val schema = new StructType().add("firstName", StringType).add("lastName", StringType)
+    val jsonDF = spark.read.schema(schema)
```

Comment:
Does schema inference work when multiLine is disabled?

Reply:
No because of many empty strings produced by Hadoop LineRecordReader. It will be fixed in separate PRs for the issues: SPARK-23725 and/or SPARK-23724 . For now you have to specify schema or use multiline mode as a temporary workaround.

Reply:
I think you should have explained this in PR description ..